lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -747,12 +762,12 @@ def indented(str)
           " " * @indent + str
         end
 
-        sig { params(method: UnboundMethod).returns(T::Boolean) }
+        sig { params(method: UnboundMethod).returns(T.nilable(T::Boolean)) }
         def method_in_gem?(method)
-          source_location = method.source_location&.first
-          return false if source_location.nil?
+          source_location, _ = method.source_location
+          return nil unless source_location
 
-          gem.contains_path?(source_location)
+          source_location == "(eval)" || gem.contains_path?(source_location)
```

Comment:
When is the source_location equal to (eval)?

Reply:
Ah, forgot to mention that in the PR description, good catch. That happens when:
```
class Foo
  module_eval("def foo; end")
end

Foo.instance_method(:foo).source_location
# => ["(eval)", 1]
```